Semver/Ranges/Comparers/Npm/NpmParseOptions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+
+namespace Semver.Ranges.Comparers.Npm
+{
+    /// <summary>
+    /// The options to use when parsing a range with npm syntax.
+    /// </summary>
+    public readonly struct NpmParseOptions : IEquatable<NpmParseOptions>
+    {
+        /// <summary>
+        /// Gets if non-explicitly selected prerelease versions should be included.
+        /// </summary>
+        public readonly bool IncludePreRelease;
+        
+        private readonly string stringValue;
```

Comment:
`ToString()` isn't going to be called very much. I would not pre-cache the string representation, but instead just compute it inside `ToString()`.

Reply:
It was done like this because it's a ```readonly struct``` so you can't change the fields outside of the constructor. I'd have to change it to a class to do this. It's probably not a problem though. I'll check.